fastestimator/pipeline/pipeline.py

```diff
@@ -330,28 +331,25 @@ def _get_batch_size(self, epoch):
     def show_results(self, inputs=None, mode="train", num_steps=1, current_epoch=0):
         data = []
         self._prepare(inputs=inputs)
-        dataset = self.dataset_schedule[mode].get_current_value(current_epoch)
-        for example in dataset.take(num_steps):
-            data.append(example)
+        ds_iter = self.dataset_schedule[mode].get_current_value(current_epoch)
+        for _ in range(num_steps):
+            data.append(next(ds_iter))
         self._reset()
         return data
 
     def benchmark(self, inputs=None, mode="train", num_steps=1000, log_interval=100, current_epoch=0):
         self._prepare(inputs=inputs)
-        dataset = self.dataset_schedule[mode].get_current_value(current_epoch)
+        ds_iter = self.dataset_schedule[mode].get_current_value(current_epoch)
         batch_size = self._get_batch_size(current_epoch)
-        num_loops = int(np.ceil(batch_size * num_steps / min(self.num_examples[mode])))
-        step = 0
-        start = time.time()
-        for _ in range(num_loops):
-            for _ in dataset:
-                step += 1
-                if step % log_interval == 0:
+        for idx in range(num_steps+1):
```

Comment:
+1 ?

Reply:
when user set num_steps = 1000 in benchmark,  we will run 0-1000(1001 steps in total). 
the 0th step is for warm up . Also by doing this, it will show the 1000th step's speed.

Reply:
like the actual _warmup() call?